src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -206,15 +216,21 @@ public IList<JsonPropertyInfo> Properties
         {
             get
             {
-                if (_properties == null)
-                {
-                    PopulatePropertyList();
-                }
-
-                return _properties;
+                _isUserModifiedPropertyList = true;
```

Comment:
won't this always be true now?

Reply:
Yes, assuming the user calls the property getter.